src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala

```diff
@@ -16,25 +16,51 @@ import scala.collection.JavaConverters._
 object Neo4jDataFrame {
 
   def mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (String,Seq[String]), relationship: (String,Seq[String]), target: (String,Seq[String])): Unit = {
+    mergeEdgeList(sc, dataFrame, source,relationship, target, Map.empty)
+  }
+
+  def mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (String,Seq[String]), relationship: (String,Seq[String]), target: (String,Seq[String]), renameMap: Map[String,String]): Unit = {
+    val sourceLbel = renameMap.getOrElse(source._2.head, source._2.head)
+    val targetLabel = renameMap.getOrElse(target._2.head, target._2.head)
     val mergeStatement = s"""
       UNWIND {rows} as row
-      MERGE (source:`${source._1}` {`${source._2.head}` : row.source.`${source._2.head}`}) ON CREATE SET source += row.source
-      MERGE (target:`${target._1}` {`${target._2.head}` : row.target.`${target._2.head}`}) ON CREATE SET target += row.target
+      MERGE (source:`${source._1}` {`${source._2.head}` : row.source.`${sourceLbel}`}) ON CREATE SET source += row.source
+      MERGE (target:`${target._1}` {`${target._2.head}` : row.target.`${targetLabel}`}) ON CREATE SET target += row.target
       MERGE (source)-[rel:`${relationship._1}`]->(target) ON CREATE SET rel += row.relationship
       """
     val partitions = Math.max(1,(dataFrame.count() / 10000).asInstanceOf[Int])
     val config = Neo4jConfig(sc.getConf)
     dataFrame.repartition(partitions).foreachPartition( rows => {
       val params: AnyRef = rows.map(r =>
         Map(
-          "source" -> source._2.map( c => (c, r.getAs[AnyRef](c))).toMap.asJava,
-          "target" -> target._2.map( c => (c, r.getAs[AnyRef](c))).toMap.asJava,
+          "source" -> source._2.map( c => (renameMap.getOrElse(c,c), r.getAs[AnyRef](c))).toMap.asJava,
```

Comment:
if this turns out to be a perf issue, we could have two lists one for source and one for target names and build it up only once.

Reply:
We're getting values out of `renamedColumns` which is a `Map`.
It's an `O[1]` operation, isn't it?